sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c

```diff
@@ -65,30 +60,29 @@ static enum gnrc_ipv6_ext_demux_status _handle_rh(gnrc_pktsnip_t *current, gnrc_
         hdr = ipv6->data;
     }
 
-    switch (ipv6_ext_rh_process(hdr, (ipv6_ext_rh_t *)ext)) {
-        case EXT_RH_CODE_ERROR:
+    switch ((res = gnrc_ipv6_ext_rh_process(hdr, (ipv6_ext_rh_t *)ext))) {
+        case GNRC_IPV6_EXT_RH_ERROR:
             /* TODO: send ICMPv6 error codes */
             gnrc_pktbuf_release(pkt);
-            return GNRC_IPV6_EXT_ERROR;
+            break;
 
-        case EXT_RH_CODE_FORWARD:
+        case GNRC_IPV6_EXT_RH_FORWARDED:
             /* forward packet */
             if (!gnrc_netapi_dispatch_receive(GNRC_NETTYPE_IPV6, GNRC_NETREG_DEMUX_CTX_ALL, pkt)) {
                 DEBUG("ipv6: could not dispatch packet to the ipv6 thread\n");
                 gnrc_pktbuf_release(pkt);
             }
-            return GNRC_IPV6_EXT_FORWARDED;
+            break;
 
-        case EXT_RH_CODE_OK:
+        case GNRC_IPV6_EXT_RH_OK:
             /* this should not happen since we checked seg_left early */
             gnrc_pktbuf_release(pkt);
-            return GNRC_IPV6_EXT_ERROR;
+            break;
     }
 
-    return GNRC_IPV6_EXT_OK;
+    return res;
```

Comment:
While technically speaking it should be impossible to reach this `return`, returning an unitialized variable that is never written here is not very pretty. Would it be okay to return `GNRC_IPV6_EXT_RH_ERROR` here as a sort of sane default?